### PR TITLE
Move clearlabels from models to issue service

### PR DESCRIPTION
**Pull request by lunny**
This PR also added a new notification type `webhook` and implemented `ClearLabels` on this notification type.

### CODE REVIEW

**Review by guillep2k — APPROVED** [review]

File: modules/notification/webhook/webhook.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+type webhookNotifier struct {
+	base.NullNotifier
+}
+
+var (
+	_ base.Notifier = &webhookNotifier{}
+)
+
+// NewNotifier create a new mailNotifier notifier
```
Comment (zeripath) [inline modules/notification/webhook/webhook.go]:
```suggestion
// NewNotifier create a new webhookNotifier notifier
```

File: services/issue/label.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// license that can be found in the LICENSE file.
+
+package issue
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/notification"
+)
+
+// ClearLabels clear an issue's all labels
```
Comment (zeripath) [inline services/issue/label.go]:
```suggestion
// ClearLabels clears all of an issue's labels
```

**Review by zeripath — APPROVED** [review]
Just a few minor comment changes.

Otherwise approve.

Does this mean that you're going to be moving all/most of the webhook calls to this new notifier?

**Comment by lunny** [comment]
@zeripath done. And yes, All my recent refactors are moving webhooks out of models and next steps I will move them to notifications.

**Pull request merged**